atomic_reactor/utils/imageutil.py

```diff
@@ -92,33 +100,76 @@ def _get_registry_client(self, registry: str) -> util.RegistryClient:
-    # and call get_inspect_for_image
-    return {}
-
-
-def remove_image(image, force=False):
-    # self.tasker.remove_image(image, force=force)
-    # most likely won't be needed at all
-    return {}
-
+        :param image: str, image pullspec from which to extract
+        :param src_path: str, path inside the image that points to file or directory
+                             that will be extracted
+        :param dst_path: str, path where to export file/dir
+        """
+        cmd = ['oc', 'image', 'extract', f'{image}', '--path', f'{src_path}:{dst_path}']
 
-def tag_image(image, new_image):
-    # self.tasker.tag_image(image, new_image)
-    return True
+        try:
+            retries.run_cmd(cmd)
+        except subprocess.CalledProcessError as e:
+            logger.error("Image file extraction failed\n%s", e.output)
+            raise
 
+    def download_image_archive_tarball(self, image: str, path: str) -> None:
+        """Downloads image archive tarball to path.
 
-def get_image(image):
-    # self.tasker.get_image(image)
-    # use skopeo copy
-    return {}
+        :param image: str, image pullspec to download
+        :param path: str, path including the filename of the tarball
+        """
+        cmd = ['skopeo', 'copy', f'docker://{image}', f'docker-archive://{path}']
+        try:
+            retries.run_cmd(cmd)
+        except subprocess.CalledProcessError as e:
+            logger.error("Image archive download failed:\n%s", e.output)
+            raise
+
+    def get_uncompressed_image_layer_sizes(self, path: str) -> List[Dict[str, Union[str, int]]]:
+        """Returns data about the uncompressed image layer sizes
+
+        :param path: str path to a image archive tarball
+        :return: List[Dict[str, object]], where key is layer id and int is the size
+                                 in bytes
+        """
+        with tarfile.open(path) as tar:
+            manifest_file = tar.extractfile('manifest.json')
+            if not manifest_file:
+                raise ValueError(f'manifest.json from {path} is not a regular file')
+            manifest = json.load(manifest_file)
+            layers = manifest[0]['Layers']
+            layer_sizes = []
+            for layer in layers:
+                layer_id = f'sha256:{layer.replace(".tar", "")}'
+                layer_sizes.append({
+                    "diff_id": layer_id,
+                    "size": tar.getmember(layer).size
+                })
+            return layer_sizes
+
+    def extract_filesystem_layer(self, src_path: str, dst_path: str) -> str:
+        """Extract filesystem layer from image archive tarball and
+        saves it at dst_path. This is meant for flatpacks and will work
+        only when the archive has 1 layer.
+
+        :param src_path: str, path to image archive tarball
+        :param dst_path: str, path where the layer will be copied
+        :return: str, the filesystem layer name
```

Comment:
Technically doesn't have to be just a name, it can be a relative path (example from the spec: `a65da33792c5187473faa80fa3e1b975acba06712852d1dea860692ccddf3198/layer.tar`)

Shouldn't be a problem though, the usage is still the same (treat the return value of this method as a relative path within dst_path)

Reply:
updated the docstring slightly